src/components/register/register.tsx

```diff
@@ -76,25 +76,39 @@ const Register = () => {
       ok = 201;
     }
     const response = await fetchJson(url, method, userData).catch(() => null);
+    let registered = false;
     if (response && response.status === ok) {
-      showMessage({ type: "success", title: "Registration successful" });
-      user.state = "Registered";
-      authService.setUser(user);
-      showMessage({
-        type: "success",
-        title: `Registration successful`,
-        detail:
-          "You have been successfully registered. To be able to login, you will need to setup two-factor authentication in the next page.",
-        label1: "Continue",
-        callback1: () => {
-          navigate("/setup-2fa");
-        },
-        modal: true,
-      });
+      // The state should has changed and the user should now have an internal ID.
+      // We need to check that and give a hint to the backend to reload the session.
+      for (let attempt = 0; attempt < 5; attempt++) {
+        const wait = (1 << attempt) * 50;
+        await new Promise(r => setTimeout(r, wait));
+        const user = await authService.getUser(true);
+        if (user?.id && user.state === 'Registered') {
+          registered = true;
+          break;
+        }
+      }
+    }
+    if (!registered) {
+      showMessage({ type: "error", title: "Could not register" });
+      back();
       return;
     }
-    showMessage({ type: "error", title: "Could not register" });
-    back();
+    showMessage({ type: "success", title: "Registration successful" });
+    user.state = "Registered";
+    authService.setUser(user);
+    showMessage({
```

Comment:
showMessage is repeated here, is it intentional?

Reply:
Was like that before. First one produces a toast message, the second one a dialog. I think the dialog is enough, so I removed it.